### PR TITLE
bpftune: init at unstable-2023-07-14, nixos/bpftune: init

### PR DESCRIPTION
**Pull request by NickCao**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/bpftune/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      --replace /sbin  /bin
+    substituteInPlace src/bpftune.service \
+      --replace /usr/sbin/bpftune "$out/bin/bpftune"
+    substituteInPlace include/bpftune/libbpftune.h \
+      --replace /usr/lib64/bpftune/       "$out/lib/bpftune/" \
+      --replace /usr/local/lib64/bpftune/ "$out/lib/bpftune/"
+  '';
+
+  nativeBuildInputs = [
+    clang
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/bpftune/default.nix:37]: Should we use clangStdenv?

**Comment by Tungsten842** [comment]
closes https://github.com/NixOS/nixpkgs/issues/240841

**Comment by jian-lin** [comment]
@NickCao Is this signature broken?
image

**Comment by NickCao** [comment]
> @NickCao Is this signature broken?

Looks so, I'm on vacation and would handle it later.

File: nixos/tests/bpftune.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  testScript = ''
+    with subtest("bpftune startup"):
+      client.wait_for_unit("bpftune.service")
+      client.wait_for_console_text("bpftune works fully")
+
+    with subtest("bpftune tcp buffer size"):
+      server.wait_for_unit("iperf3.service")
+      client.succeed("iperf3 --reverse -c server")
+      client.wait_for_console_text("need to increase TCP buffer size")
```
[the remark this answers is not in the document]
Comment (NickCao) [reply nixos/tests/bpftune.nix:34]: We can always update the tests to account for later changes.

**Comment by SuperSandro2000** [comment]
The test times out on aarch64

**Pull request merged**